src/metadata.cc

```diff
@@ -0,0 +1,31 @@
+#include "metadata.hh"
+#include "toml.hpp"
+#include <QDebug>
+
+std::optional< metadata > loadMetadata( std::string_view filepath )
+{
+  // by default, the optional will be initialized as std::nullopt
+  metadata result{};
+  toml::table tbl;
+
+  try {
+    tbl = toml::parse_file( filepath );
+  }
+  catch ( toml::parse_error & e ) {
+    qWarning() << "Failed to get Metadata for:" << filepath << "with error:" << e.description();
+    return std::nullopt;
+  }
+
```

Comment:
an extra std::exception？ to avoid crash?  file not exited?

Reply:
toml::parse_error which is a subclass of std::runtime_error already includes that
![image](https://user-images.githubusercontent.com/20123683/235334497-15f4abc1-204e-4f7c-9755-05817dd83c76.png)